app/controllers/pledgegroups_controller.rb

```diff
@@ -0,0 +1,46 @@
+class PledgegroupsController < ApplicationController
+  def index
+    @pledgegroups = Pledgegroup.all
+  end
+
+  def new
+    @pledgegroup = Pledgegroup.new
+  end
+
+  
+  def create
+    @pledgegroup = Pledgegroup.new(pledgegroup_params)
+
+    if @pledgegroup.save
+      redirect_to pledgegroups_url
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
```

Comment:
```suggestion
  def create
    @pledgegroup = Pledgegroup.new({ budget: params[:pledgegroup][:budget], admins: [current_user], is_public: params[:is_public] })

    if @pledgegroup.save
      redirect_to pledgegroups_url
    else
      flash[:error] = "Pledge group could not be saved with the given input"
      redirect_back fallback_location: new_pledgegroup_path
    end
  end
```
Gets rid of the extra private methods.

also invokes redirect_back instead of render so that there's a chance that user-provided input can be saved.